### ISSUE TITLE
postgres_to_exasol.sql doesn't appear to migrate anything

### ISSUE THREAD

**Issue by hamx0r** [EXASOL/database-migration#6]
Using Exasol 6.2.1 and postgres 9.6.11 (AWS RDS), I can run the [postgres_to_exasol.sql](https://github.com/exasol/database-migration/blob/a02eefc2f03446c5bc48eade9c83c746ca0f8c59/sqlserver_to_exasol.sql) script after changing the `create connection postgres_db` line to match my environment, and the following text is displayed:

```
[2019-09-19 14:26:55] Run migrate_postgres_to_exasol.sql
[2019-09-19 14:26:55] Connecting to Exasol BYOL...
[2019-09-19 14:26:57] Using batch mode (1000 insert/update/delete statements max)
create schema if not exists database_migration
[2019-09-19 14:26:57] 0 row(s) affected in 113 ms
create or replace script database_migration.POSTGRES_TO_EXASOL(
CONNECTION_NAME              -- name of the database connection inside exasol -> e.g. postgres_db
,IDENTIFIER_CASE_INSENSITIVE -- true if identifiers should be stored case-inse...
[2019-09-19 14:26:57] 0 row(s) affected in 254 ms
create connection postgres_db to 'jdbc:postgresql://{my connection string here}
[2019-09-19 14:26:57] 0 row(s) affected in 133 ms
execute script database_migration.POSTGRES_TO_EXASOL(
'postgres_db', -- name of your database connection
true,          -- case sensitivity handling for identifiers -> false: handle them case sensitiv / true: handle them case insens...
[2019-09-19 14:27:02] completed in 4 s 430 ms
[2019-09-19 14:27:02] Summary: 4 of 4 statements executed in 6 s 305 ms (12011 symbols in file)
```

Here we see it completed in 4s, and when I look at my Exasol DB, I don't see any new tables imported from Postgres.   Is there something else I should be doing to make this work?  I've verified that the Postgres user in the connection string has access to `information_schema.columns`, so I believe the script can introspect properly.

**Comment by exaValerieWiedemann** [comment]
Hello Jason,
The script database_migration.POSTGRES_TO_EXASOL generates the DDL and import statement, but does not execute them. It returns these statements as an output, so you need to copy these statements and execute them.
What kind of client did you use to execute the script? It looks like you're working with a command line version. You can have a look at the section SQL clients in the Exasol user manual to choose one, for example the open source tool DBeaver.
https://docs.exasol.com/connect_exasol/sql_clients.htm
Let us know if that helps,
Valerie

**Comment by hamx0r** [comment]
Thanks, @exaValerieWiedemann.  I was using Pycharm's built-in DB tool since it (mostly?) works with Exasol.  Upon using DBeaver, I'm now getting the output you mentioned:  When running the migration script, it generates 3 SQL statements I can use:

1. Create schema
2. Create table
3. Import 

I'm running the import now, and the fact it's taking more than 6 seconds gives me hope!

**Comment by hamx0r** [comment]
@exaValerieWiedemann I wish to migrate some views in Postgres to an Exasol table.  I am not allowed to read the source table, but I am allowed to read the source view.  How could I tweak the import script to read a Postgres view and import it to Exasol as a table with the same name (ie `v_table_view`)?

**Comment by hamx0r** [comment]
Sorry to be a be a bother, but I'm noticing that even with DBeaver, this script sometimes returns nothing.  Does this happen if the DB is undergoing other operations?  For example, have begun importing data to Exasol from Postgres, and while importing, I'm executing the `POSTGRES_TO_EXASOL` script so that it can generate SQL I can use to do more table creation and imports later.  In this circumstance, the response is just `EMPTY`.  Is this expected behavior or is something else wrong?

**Comment by exaValerieWiedemann** [comment]
Hi Jason,
regarding the view:
You could write an import statement that reads from the view instad of the table. It's also possible to modify the select-statement that starts in line 26 of the script so that it also creates the DDL for views and not only tables.
An empty result is not the expected behaviour, this usually only happens if you call it on a schema/table that does not exist or is written in the wrong case (e.g. lower case instead of upper case). Does this even happen if you call it with the exact same schemas and tables? if yes, you could examine which statements the postgres database gets by having a look at the auditing data on that end. Maybe that way you can identify the cause?

**Comment by hamx0r** [comment]
Thanks for the tip on Views.  

I found that several tools will, for a while, not properly display SQL when running the migration script.  It seems to be when a migration task is already running, or perhaps when Exasol is doing some background task (indexing?  not sure).  I found the PyCharm, DBeaver, and dBVisualizer all can work (ie I run migration script and get back 3 rows of SQL to run), but some cases, the same queries return nothing.  I'll go ahead and close this since the solution seems to merely be patience.
